### DISCUSSION TITLE
Can I "extend" Activity model to use a Trait?

### DISCUSSION THREAD

**Discussion by hmaesta** [spatie/laravel-activitylog#959] [General] [answered]
My project uses a **Trait** to [convert dates into user's defined timezone](https://github.com/DivineOmega/dates-timezone-conversion-trait/blob/master/src/Traits/DatesTimezoneConversion.php).

For every Model that has dates we just need to point this Trait and specify date fields into `$dates`. 
Super simple and works fine 👌

But all activity logs return dates in UTC – like it should, since I didn't pointed the Trait in model.

I know I can **_overwrite_** Model path in [`activity_model`](https://github.com/spatie/laravel-activitylog/blob/1d8acdac3e1f3d6937f8d10fd005de71b0350bc4/config/activitylog.php#L38), but I don't want to copy file content and take the risk of missing future updates in that file.

So, is it possible to ""extend"" this Model from source and add my Trait?

```php
use MyTraitName;

protected $dates = [
   'created_at',
   'updated_at',
];
```

_Ps:_ Laravel 8 + PHP 7.4 + laravel-activitylog 3.17

**Comment by Gummibeer** [comment, marked as answer]
You can extend it without writing your whole own model.

```php
class MyActivity extends SpatieActivity
{
    use DateTrait;

    protected $dates = [
       'created_at',
       'updated_at',
    ];
}
```

**Reply by hmaesta** [reply]
Thanks, Tom! 🙌
I know see that this was a silly question, but I actually didn't know that was possible.

---

#### For future Googlers 👇  

1. Create new Model: `php artisan make:model Activity`
2. Edit `activity_model` in `config/acitivitylog.php` 
```diff
- 'activity_model' => \Spatie\Activitylog\Models\Activity::class,
+ 'activity_model' => \App\Models\Activity::class,
```
3. Edit `app/Models/Activity.php` to extend `SpatieActivity`
```php
namespace App\Models;

use App\Traits\DateTrait;
use Spatie\Activitylog\Models\Activity as SpatieActivity;

class Activity extends SpatieActivity
{
    use DateTrait;

    protected $dates = [
        'created_at',
        'updated_at'
    ];
}
```

4. Everywhere you are calling Activity, replace default Model import by yours.
```diff
- use Spatie\Activitylog\Models\Activity;
+ use App\Models\Activity;
```

**Reply by tinusg** [reply]
Exactly what I was looking for.
